LadybugTools_Engine/Python/src/ladybugtools_toolkit/external_comfort/comfort_feasibility.py

```diff
@@ -0,0 +1,715 @@
+                        surface_temperatures=epw.dry_bulb_temperature,
+                    ).mean_radiant_temperature
+                    with open(mrt_unshaded_file, "w", encoding="utf-8") as fp:
+                        json.dump(mrt_unshaded.to_dict(), fp)
+                if not mrt_shaded_file.exists():
+                    mrt_shaded = mrt_unshaded.get_aligned_collection(
+                        epw.dry_bulb_temperature.values
+                    )
+                    with open(mrt_shaded_file, "w", encoding="utf-8") as fp:
+                        json.dump(mrt_shaded.to_dict(), fp)
+                for am in additional_air_moistures:
+                    # calculate DBT and RH components - included here to allow parallelisation
+                    dbt_file = out_dir / f"DBT_{am:0.1f}.json"
+                    rh_file = out_dir / f"RH_{am:0.1f}.json"
+                    if not dbt_file.exists() or not rh_file.exists():
+                        dbt, rh = evaporative_cooling_effect_collection(
+                            epw=epw, evaporative_cooling_effectiveness=am
+                        )
+                        with open(dbt_file, "w", encoding="utf-8") as fp:
+                            json.dump(dbt.to_dict(), fp)
+                        with open(rh_file, "w", encoding="utf-8") as fp:
+                            json.dump(rh.to_dict(), fp)
+                    for wm in wind_multipliers:
+                        if tci.value == ThermalComfortIndex.UTCI.value:
+                            target_iterations.append(
+                                {
+                                    "epw": epw,
+                                    "thermal_comfort_index": tci,
+                                    "shade_proportion": sp,
+                                    "additional_air_moisture": am,
+                                    "wind_multiplier": wm,
+                                    "met_rate": None,
+                                    "clo_value": None,
+                                }
+                            )
+                        else:
+                            for mr in met_rates:
+                                for cv in clo_values:
+                                    target_iterations.append(
+                                        {
+                                            "epw": epw,
+                                            "thermal_comfort_index": tci,
+                                            "shade_proportion": sp,
+                                            "additional_air_moisture": am,
+                                            "wind_multiplier": wm,
+                                            "met_rate": mr,
+                                            "clo_value": cv,
+                                        }
+                                    )
+    # shuffle the iterations
+    np.random.shuffle(target_iterations)
```

Comment:
Possibly so that if a partial study is run, better distributed coverage is achieved? Although limiting study length currently doesn't seem to be implemented. Shouldn't be doing any harm leaving it in.

Reply:
I agree that it doesn't need changing, was just a bit confused when initially reading through the code and wished for clarification!

Reply:
I can't remember why I specifically shuffled iterations here, but in the past this is what I've done for particularly long processes where a file was generated as part of the output - and the process could be parallelised but I was too lazy to get it running properly 😄. 

I'd run several instances of Python iterating over shuffled lists and generating output, and each instance would check to see if a file exists and skip that iteration if it did. Not the best way of doing things I know, but it saved a few headaches with ProcessPoolExecutor in the past as it would just bypass the need to restructure the code and instead just brute-force the whole process with lower risk of race conditions. 

In this case it was probably a line that crept into the code after I pasted it from this kind of thing elsewhere. It can be removed if needed and it wouldn't matter.